### PR TITLE
Updated numpy easyblock to use read_file for patch.

### PR DESCRIPTION
**Pull request by dithwick**
This fixes issue #2393 (for numpy.py only).

### CODE REVIEW

File: easybuild/easyblocks/n/numpy.py [hunk 1]
```diff
@@ -142,7 +142,7 @@ def get_libs_for_mkl(varname):
                 # patches are either strings (extension) or dicts (easyblock)
                 if isinstance(patch, dict):
                     patch = patch['path']
-                if patch_wl_regex.search(open(patch, 'r').read()):
+                if patch_wl_regex.search(read_file(patch)):
```
Comment (smoors) [inline easybuild/easyblocks/n/numpy.py:145]: hm, it's probably me but I don't see how this fixes it? doesn't the `read_file` function do essentially the same?

Comment (smoors) [reply easybuild/easyblocks/n/numpy.py:145]: @dithwick can you create a test report that includes this PR?

Comment (dithwick) [reply easybuild/easyblocks/n/numpy.py:145]: Yep will do!

**Comment by dithwick** [comment]
Test report by @dithwick
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _SciPy-bundle-2020.11-intel-2020b.eb_ 

Build succeeded for 1 out of 1 (1 easyconfigs in total)
moorcroft.scrtp.warwick.ac.uk - Linux centos linux 7.9.2009, x86_64, Intel(R) Core(TM) i5-8600 CPU @ 3.10GHz, Python 2.7.5
See https://gist.github.com/7d3643dd0a71a8a7640799e63c28cfef for a full test report.

**Review by smoors — APPROVED** [review]
lgtm

**Comment by smoors** [comment]
Going in, thanks @dithwick!

**Pull request merged**